src/integrations/prefect-kubernetes/prefect_kubernetes/events.py

```diff
@@ -51,62 +45,52 @@ def __init__(
         worker_resource["prefect.resource.role"] = "worker"
         worker_related_resource = RelatedResource(worker_resource)
         self._related_resources = related_resources + [worker_related_resource]
-
-        self._watch = kubernetes.watch.Watch()
-        self._thread = threading.Thread(target=self._replicate_pod_events)
-
         self._state = "READY"
 
-        atexit.register(self.stop)
-
-    def __enter__(self):
-        """Start the replicator thread."""
-        self._thread.start()
+    async def __aenter__(self):
+        """Start the Kubernetes event watcher when entering the context."""
+        self._task = asyncio.create_task(self._replicate_pod_events())
         self._state = "STARTED"
+        return self
 
-    def __exit__(self, *args, **kwargs):
-        """Stop the replicator thread."""
-        self.stop()
-
-    def stop(self):
-        """Stop watching for pod events and stop thread."""
-        if self._thread.is_alive():
-            self._watch.stop()
-            self._thread.join()
-            self._state = "STOPPED"
+    async def __aexit__(self, exc_type, exc_value, traceback):
+        """Stop the Kubernetes event watcher and ensure all tasks are completed before exiting the context."""
+        self._state = "STOPPED"
+        if self._task:
+            await self._task
 
     def _pod_as_resource(self, pod: "V1Pod") -> Dict[str, str]:
         """Convert a pod to a resource dictionary"""
         return {
             "prefect.resource.id": f"prefect.kubernetes.pod.{pod.metadata.uid}",
-            "prefect.resource.name": pod.metadata.name,
-            "kubernetes.namespace": pod.metadata.namespace,
+            "prefect.resource.name": str(pod.metadata.name),
```

Comment:
The way that `kubernetes_asyncio` gets generated, none of the api methods in clients  are async but return coroutines. So when we mock the clients we have it so that they return an async mock.

For `_pod_as_resource` we cast them to strings due to a vaildation error during testing because of mocks.

Reply:
I feel like in general its not ideal to change the actual code because of how tests work, but in this case i dont think its a big deal